src/ApiHandler.ts

```diff
@@ -361,47 +361,69 @@ export default class ApiHandler {
 		}
 	}
 
+	/**
+	 *
+	 * @param hash BlockHash to make call at.
+	 * @param stash: _Stash_ address.
+	 */
 	// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-	async fetchStakingLedger(hash: BlockHash, address: string) {
+	async fetchStakingInfo(hash: BlockHash, stash: string) {
 		const api = await this.ensureMeta(hash);
 
-		const [header, staking] = await Promise.all([
+		const [header, controllerOption] = await Promise.all([
 			api.rpc.chain.getHeader(hash),
-			api.query.staking.ledger.at(hash, address),
+			api.query.staking.bonded.at(hash, stash), // Option<AccountId> representing the controller
 		]);
 
 		const at = {
 			hash,
 			height: header.number.toNumber().toString(10),
 		};
 
-		if (staking.isNone) {
-			// Address is not a Controller, no need to look up slashing spans.
-			return {
-				at,
-				staking: {},
-				numSlashingSpans: null,
+		if (
+			controllerOption.isNone === true ||
+			controllerOption.isSome !== true
+		) {
+			// if controllerOption.isNone, `stash` address param is not a stash.
+			throw {
+				error: `The address ${stash} is not a stash address.`,
+				statusCode: 422,
```

Comment:
The simplest solution is usually the correct one.
```suggestion
				statusCode: 400,
```